File: teste.py
cod_otto_interesse = '86289161'

#   método que seleciona os trechos a jusante do ponto de interesse
rios = ['','','','','','']
n_rio = 0

# seleção dos rios a jusante
for valor in range(len(cod_otto_interesse)):
    index_analise = (valor + 1) * -1
    algarismo = int(cod_otto_interesse[index_analise])
    impar = algarismo % 2
    codf = cod_otto_interesse[:(len(cod_otto_interesse)-valor)]

    # fiz o teste para final par, pois do contrário dá erro, não entendi
    if impar == 0:
        codfim = cod_otto_interesse[:(len(cod_otto_interesse)-valor)]
        rios[n_rio] = codfim
        n_rio = n_rio + 1
        compri = len(codfim)
        if compri <= 4:    # teste para ver se no rio princila da bacia (rio 8628)
            break

print('n_rio:', n_rio)
print('rios:', rios)

selecao = ''
for elementos in rios:
    if elementos != '':
        selecao = selecao + 'camada_ottotrechos.cocursodag = \''+ elementos +'\' OR '
    else:
        break
comp = len(selecao)
comp2 = comp - 3
print('selecao:', selecao)
sele2 = selecao [:comp2]
print('sele2:', sele2)
query_ottotrechos_jusante = '?query=SELECT camada_ottotrechos.cocursodag, camada_ottotrechos.geometry '\
                            'FROM camada_ottotrechos '\
                            'WHERE ('+ sele2 +')AND camada_ottotrechos.cobacia < \''+cod_otto_interesse+'\''
print('query_ottotrechos_jusante:', query_ottotrechos_jusante)


print('\n''-> Seleção de ottorechos a jusante realizado.')

File: teste.js
let cod_otto_interesse = '86289161';

// Método que seleciona os trechos a jusante do ponto de interesse
let rios = ['', '', '', '', '', ''];
let n_rio = 0;

// Seleção dos rios a jusante
for (let valor = 0; valor < cod_otto_interesse.length; valor++) {
    let index_analise = (valor + 1) * -1;
    let algarismo = parseInt(cod_otto_interesse[index_analise]);
    let impar = algarismo % 2;
    let codf = cod_otto_interesse.slice(0, cod_otto_interesse.length - valor);

    // Teste para final par
    if (impar === 0) {
        let codfim = cod_otto_interesse.slice(0, cod_otto_interesse.length - valor);
        rios[n_rio] = codfim;
        n_rio++;
        let compri = codfim.length;

        // Teste para ver se no rio principal da bacia (rio 8628)
        if (compri <= 4) {
            break;
        }
    }
}

console.log('n_rio:', n_rio);
console.log('rios:', rios);

let selecao = '';
for (let elementos of rios) {
    if (elementos !== '') {
        selecao += 'cocursodag = \'' + elementos + '\' OR ';
    } else {
        break;
    }
}

let comp = selecao.length;
let comp2 = comp - 3;
console.log('selecao:', selecao);
let sele2 = selecao.slice(0, comp2);
console.log('sele2:', sele2);
let consulta = '(' + sele2 + ') AND cobacia_n < \'' + cod_otto_interesse + '\'';
console.log('consulta:', consulta);

console.log('\n-> Seleção de ottorechos a jusante realizado.');
